robot/impl/local_robot_test.go

```diff
@@ -3547,4 +3538,72 @@ func TestMachineStatus(t *testing.T) {
 		)
 		rtestutils.VerifySameResourceStatuses(t, mStatus.Resources, expectedStatuses)
 	})
+
+	t.Run("poll during reconfiguration", func(t *testing.T) {
+		rev1 := "rev1"
+		lr := setupLocalRobot(t, ctx, &config.Config{
+			Revision:   rev1,
+			Components: []resource.Config{newMockConfig("m", 200, false, "")},
+		}, logger)
+
+		// update resource with a working config that is slow to reconfigure.
+		rev2 := "rev2"
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			lr.Reconfigure(ctx, &config.Config{
+				Revision:   rev2,
+				Components: []resource.Config{newMockConfig("m", 300, false, "1s")},
+			})
+		}()
+		// sleep for a short amount of time to allow the machine to receive a new
+		// revision. this sleep should be shorter than the resource update duration
+		// defined above so that updated resource is still in a "configuring" state.
+		time.Sleep(time.Millisecond * 100)
```

Comment:
relying on a sleep is a little janky - alternatively we could poll machine status until the revision on the machine is updated and fail after some attempts. any other suggestions are welcome here.

Reply:
can do a `WaitForAssertion`